Feat : Added instructions for running testnet node in Windows

HariniRajan397 opened the pull request:
…miner node) on Windows using Binary.

Fixes blockstack/docs#661

## Description
Added instructions for running the testnet node (Both follower and miner node) on Windows from the binary available. 

1. Motivation for change
Most of the docs are Linux users oriented and hence added instructions for Windows users. 

3. How does this impact application developers
This helps and motivates Windows users to get started with their testnet node. 

4. Link to relevant issues and documentation

https://github.com/blockstack/docs/issues/661 - This is the relevant issue and the refernce is from 
https://github.com/blockstack/docs/issues/661#issuecomment-723516837 as well as from https://discordapp.com/channels/621759717756370964/728030782693834862/774786060738887681.


## Type of Change
- [x] New feature

## Does this introduce a breaking change?
No

- [ ] Link to documentation updates: 

## Testing information

Provide context on how tests should be performed.

1. Is testing required for this change?
 This is tested on Windows and if needed, one more round of testing in a fresh Windows machine can be done. 

## Checklist
@agraebe @whoabuddy @zrixes

HariniRajan397 commented:
@whoabuddy and @zrixes I have added teh instructions for Windows and created a pull request. Sincere thanks to both of you !

agraebe commented on src/pages/mining.md:
````diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
 ```
 
+## Running a miner in Windows
+
````
are there any requirements that we should point out?

HariniRajan397 replied:
Updated the requirements.

agraebe commented on src/pages/mining.md:
```diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
+
+To setup the miner, first, we need to generate a keychain. With this keychain, we'll get some testnet BTC from a faucet, and then use that BTC to start mining.
+
+To get a keychain, the simplest way is to use the `stacks-cli`.We'll use the `stx make-keychain` command,and pass -t to indicate that we want a testnet keychain.
```
"use the `stacks-cli`. We'll use the" (spacing after dot). makes me think: could you please run a linter on this file and fix linting issues?

agraebe replied:
"command,and pass -t" > "command, and pass `-t`" (spacing and code highlight)

HariniRajan397 replied:
Fixed the same .

agraebe commented on src/pages/mining.md:
```diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
+
+Request BTC from faucet:
+
+We need to get some testnet BTC to that address. Grab the btcAddress field, and call the BTC faucet:
```
btcAddress > `btcAddress`

HariniRajan397 replied:
Fixed it.

agraebe commented on src/pages/mining.md:
````diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
+
+```bash
+# replace <btc_address> with `btcAddress` property from your keychain
+curl -XPOST "https://stacks-node-api.blockstack.org/extended/v1/faucets/btc?address=<btc_address>"
````
I think I remember someone struggling with curl on windows - could we add it as a req and link to a place to install it on windows?

HariniRajan397 replied:
Agree. Added the appropriate links to install curl on Windows.

agraebe commented on src/pages/mining.md:
```diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
+
+### Create configuration file
+
+Now, we need to configure our node to use this Bitcoin keychain. In the `folder where your binary is extracted`, create a new file called `testnet-miner-conf.toml`.
```
`folder where your binary is extracted` > folder where your binary is extracted

(we should use `var` for code/var highlights mostly)

HariniRajan397 replied:
Fixed it.

agraebe commented on src/pages/mining.md:
````diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
+```bash
+npm install --global @stacks/cli
+stx make_keychain -t > cli_keychain.json
+type cli_keychain.json
````
so, if I read this correctly, this is the most divergence from the non-windows steps, correct? if it is just that - we should rather make a note in the other tutorial and avoid duplicating 90% of content from the other tutorial. wdyt @HariniRajan397 ?

agraebe commented on src/pages/stacks-blockchain/running-testnet-node.md:
```diff
@@ -91,6 +91,51 @@ INFO [1588108047.585] [src/chainstate/stacks/index/marf.rs:732] First-ever block
+
+First, download the binary using the below link:
+
+<https://github.com/blockstack/stacks-blockchain/releases/download/v23.0.0.10-krypton/windows-x64.zip>
```
I'd rather do this: "[download the binary](https://github.com/blockstack/stacks-blockchain/releases/download/v23.0.0.10-krypton/windows-x64.zip)"

HariniRajan397 replied:
Agreed. Updated the details along with the note.

agraebe commented on src/pages/stacks-blockchain/running-testnet-node.md:
```diff
@@ -91,6 +91,51 @@ INFO [1588108047.585] [src/chainstate/stacks/index/marf.rs:732] First-ever block
+
+<https://github.com/blockstack/stacks-blockchain/releases/download/v23.0.0.10-krypton/windows-x64.zip>
+
+Extract the binary .Open the command prompt from the folder where binary is extracted and execute the below command:
```
"binary .Open" > "binary. Open" (spacing)

-> again, would run a linter. I would assume it would fail with the current sources and we'd want to fix these issues. Failing lint tests would break the deployment of the PR

HariniRajan397 replied:
Fixed the linting issues.

agraebe reviewed (commented):
thanks for submitting the PR. Windows instructions will be very useful, so this is very impactful. I left some notes.

tl;dr: fix linting issues & explorer if we can put windows-specific (divergent) steps into the existing tutorial (most of the content seems very similar to non-windows instructions)

HariniRajan397 reviewed (commented):
Incorporated the review comments and updated the repo.

CharlieC3 reviewed (commented):
Just tested these instructions out on my Windows machine and they worked great! I followed the instructions to the letter for both running a follower node and a miner, and doing that helped me find a couple areas where users may need some extra assistance.

1. When installing MS BuildTools, the C++ package is not selected by default. It might be worth adding a small excerpt next to this prereq telling the user they'll need to manually select the C++ build tools option when installing.

1. After first starting a follower node, I received the below Windows Defender warning. We should add an excerpt explaining the user may see a Windows Defender warning when first running the node, and to be sure to select "Allow Access" when/if prompted. This excerpt should be added to both the testnet and miner pages.

HariniRajan397 commented:
Thanks, @CharlieC3 ! Incorporated the review comments and added the instructions with an accompanying snapshot for the C++ Build tools as well as Windows Defender popup. Kindly re-review the latest commit 
[c64b4b6 ](https://github.com/blockstack/docs/pull/891/commits/c64b4b6aaaeb5d8716e43d87201cec86b10d5bae). Thanks.

cc @agraebe

agraebe reviewed (approved):
looks great, thanks @HariniRajan397 !

CharlieC3 reviewed (approved):
I like the additions you made! :shipit:
